conslogging/conslogging.go

```diff
@@ -467,11 +468,27 @@ func (cl ConsoleLogger) color(c *color.Color) *color.Color {
 	return noColor
 }
 
+var (
+	// githubRegExp Matches :2dd88e53f2e59e96ec1f9215f24a3981e5565edf+ in a target.
+	// 	Prefix containing hash may resemble: g/e/hello-world:2dd88e53f2e59e96ec1f9215f24a3981e5565edf+base
+	githubRegExp = regexp.MustCompile(`:[a-zA-Z0-9]*\\+`)
```

Comment:
this feels too loose; it could match `g/e/hello-world:feedfacecafe`, which isn't guaranteed to be a git sha.

instead we should match on exactly 40 chars.

```suggestion
	githubRegExp = regexp.MustCompile(`:[a-zA-Z0-9]{40}+`)
```

Reply:
I've done just that - and also found pre-existing bugs handling branch names with special characters.
Great push for tests ended with an extra bug fix :)

Reply:
thanks for adding the tests! nice find with the extra bug.

I think it would be useful to add a test that has a branch name that matches `a-f`, such as the original example I gave `feedfacecafe`.

I also just noticed that the regex should be `a-f` and not `a-z`, since a git sha is encoded as a hexidecimal.

Reply:
I've added that test now as well